device/src/async_device/mod.rs

```diff
@@ -22,28 +22,74 @@ use lorawan::{
 use rand_core::RngCore;
 
 type DevNonce = lorawan::parser::DevNonce<[u8; 2]>;
-use crate::radio::types::RadioBuffer;
 pub use crate::region::DR;
+use crate::{private::Sealed, radio::types::RadioBuffer, GetRng};
 pub mod radio;
 use core::cmp::min;
 
+/// Type-level version of the [`None`] variant
+pub struct NoneT;
```

Comment:
What about just using the unit type '()'  for this?

Reply:
Sure, I can change it back to`()` if you prefer, obviously functionally it makes no difference. We use `NoneT` over at atsamd-hal, admittedly because the code is very heavy on type-level stuff. However, to me, looking at something like `Device<C, R, T, NoneT, N>` makes it instantly clear that the empty type is deliberate. Maybe it's just me, but `Device<C, R, T, (), N>` feels a little more ambiguous.

Reply:
That's fair enough, I didn't think too much about that.